### PR TITLE
Define toolkit_initModule metric

### PR DESCRIPTION
**Pull request by rbbarad**
Define toolkit_initModule metric. This metric will help us identify issues that arise during each step of the toolkit initialization process.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by awschristou — APPROVED** [review]

File: telemetry/definitions/commonDefinitions.json [hunk 1]
```diff
@@ -5959,6 +5964,36 @@
...
+                {
+                    "type": "attempts",
+                    "required": false
+                },
+                { 
+                    "type": "duration",
+                    "required": false
+                },
+                { 
+                    "type": "errorCode",
```
Comment (shruti0085) [inline telemetry/definitions/commonDefinitions.json:5980]: Reason, errorCode is already part of our error metadata optional fields and does not need to be specified here

Comment (rbbarad) [reply telemetry/definitions/commonDefinitions.json:5980]: ah thanks for catching that - i'll remove those

File: telemetry/definitions/commonDefinitions.json [hunk 2]
```diff
@@ -1432,6 +1432,11 @@
             ],
             "description": "The type of an SQS Queue"
         },
+        {
+            "name": "startUpComponent",
+            "type": "string",
+            "description": "The Toolkit component being initialized."
```
Comment (justinmk3) [inline telemetry/definitions/commonDefinitions.json:1438]: there is already a "component" field https://github.com/aws/aws-toolkit-common/blob/b362bc7c5b65f12fee154b1c17ca896e5645cafc/telemetry/definitions/commonDefinitions.json#L819-L830

can it be re-used. if you need a free-form string, just cast it.

File: telemetry/definitions/commonDefinitions.json [hunk 3]
```diff
@@ -5959,6 +5964,28 @@
                 }
             ]
         },
+        {
+            "name": "toolkit_componentInit",
```
Comment (justinmk3) [inline telemetry/definitions/commonDefinitions.json:5968]: most metrics are named with the "verb" first. also, `toolkit_init` exists so naming this similarly helps discoverability.

```suggestion
            "name": "toolkit_initComponent",
```

File: telemetry/definitions/commonDefinitions.json (hunk 2 above)
Comment (rbbarad) [reply telemetry/definitions/commonDefinitions.json]: defined a new "module" generic metadata to give us future reusability

File: telemetry/definitions/commonDefinitions.json (hunk 3 above)
Comment (rbbarad) [reply telemetry/definitions/commonDefinitions.json]: renamed to "toolkit_initModule" with its accompanying "module" metadata

File: telemetry/definitions/commonDefinitions.json [hunk 4]
```diff
@@ -1260,6 +1260,11 @@
             "type": "string",
             "description": "User locale. Examples: en-US, en-GB, etc."
         },
+        {
+            "name": "module",
+            "type": "string",
+            "description": "A generic module metadata"
```
Comment (justinmk3) [inline telemetry/definitions/commonDefinitions.json:1266]: Example values will help devs understand how/when to use this

Comment (rbbarad) [reply telemetry/definitions/commonDefinitions.json:1266]: Updated in https://github.com/aws/aws-toolkit-common/pull/779

**Pull request merged**